### PR TITLE
enhance: rename tokenizer to analyzer and check analyzer params

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/35853

### CODE REVIEW

File: internal/core/src/common/FieldMeta.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@
 namespace milvus {
 TokenizerParams
```
Comment (zhengbuqian) [inline internal/core/src/common/FieldMeta.cpp:21]: rename method and type name as well: `TokenizerParams` and `ParseTokenizerParams`

Comment (aoiasd) [reply internal/core/src/common/FieldMeta.cpp:21]: next pr, all tokenizer in segcore and go code shoud rename to analyzer, but this pr just rename user params.

File: internal/util/function/bm25_function.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ type BM25FunctionRunner struct {
 
 func getTokenizerParams(field *schemapb.FieldSchema) string {
 	for _, param := range field.GetTypeParams() {
-		if param.Key == "tokenizer_params" {
+		if param.Key == "analyzer_params" {
```
Comment (czs007) [inline internal/util/function/bm25_function.go:45]: how about define a Const in common.

**Comment by aoiasd** [comment]
relate: https://github.com/milvus-io/milvus/issues/36751

File: internal/util/ctokenizer/text_schema_validator.go [hunk 3]
```diff
@@ -17,14 +17,14 @@ import (
...
 	h := typeutil.CreateFieldSchemaHelper(fieldSchema)
-	if !h.EnableMatch() {
+	if !h.EnableMatch() && !EnableBM25 {
 		return nil
 	}
 
-	if !h.EnableTokenizer() {
-		return fmt.Errorf("field %s is set to enable match but not enable tokenizer", fieldSchema.Name)
+	if !h.EnableAnalyzer() {
+		return fmt.Errorf("field %s is set to enable match but not enable analyzer", fieldSchema.Name)
```
Comment (czs007) [inline internal/util/ctokenizer/text_schema_validator.go:27]: It may not only enable matching but also activate BM25.

**Comment by aoiasd** [comment]
rerun cpp-unit-test

File: internal/distributed/proxy/httpserver/handler_v2.go [hunk 4]
```diff
@@ -1321,7 +1321,12 @@ func (h *HandlersV2) createCollection(ctx context.Context, c *gin.Context, anyRe
 				}
 			}
 			for key, fieldParam := range field.ElementTypeParams {
-				fieldSchema.TypeParams = append(fieldSchema.TypeParams, &commonpb.KeyValuePair{Key: key, Value: fmt.Sprintf("%v", fieldParam)})
+				value, err := getElementTypeParams(fieldParam)
+				if err != nil {
+					return nil, err
+				}
+				log.Info("test--- ", zap.String("key", key), zap.String("value", value))
```
Comment (czs007) [inline internal/distributed/proxy/httpserver/handler_v2.go:1328]: remember to delete this

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun cpp-unit-test

File: internal/distributed/proxy/httpserver/utils.go [hunk 5]
```diff
@@ -1510,6 +1510,18 @@ func convertToExtraParams(indexParam IndexParam) ([]*commonpb.KeyValuePair, erro
...
 }
 
+func getElementTypeParams(param interface{}) (string, error) {
+	if str, ok := param.(string); ok {
+		return str, nil
+	}
+
+	jsonBytes, err := json.Marshal(param)
+	if err != nil {
+		return "nil", err
```
Comment (czs007) [inline internal/distributed/proxy/httpserver/utils.go:1520]: what does "nil" mean for?

**Pull request merged**